src/test/java/team/silvertown/masil/masil/service/MasilServiceTest.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+import team.silvertown.masil.common.exception.DataNotFoundException;
+import team.silvertown.masil.common.map.KakaoPoint;
+import team.silvertown.masil.masil.domain.Masil;
+import team.silvertown.masil.masil.domain.MasilPin;
+import team.silvertown.masil.masil.dto.CreatePinRequest;
+import team.silvertown.masil.masil.dto.CreateRequest;
+import team.silvertown.masil.masil.dto.CreateResponse;
+import team.silvertown.masil.masil.exception.MasilErrorCode;
+import team.silvertown.masil.masil.repository.MasilPinRepository;
+import team.silvertown.masil.masil.repository.MasilRepository;
+import team.silvertown.masil.user.domain.ExerciseIntensity;
+import team.silvertown.masil.user.domain.Provider;
+import team.silvertown.masil.user.domain.Sex;
+import team.silvertown.masil.user.domain.User;
+import team.silvertown.masil.user.repository.UserRepository;
+
+@SpringBootTest
+@Transactional
+@DisplayNameGeneration(ReplaceUnderscores.class)
+class MasilServiceTest {
+
+    static final Faker faker = new Faker(Locale.KOREA);
+    static double dongAnLat = 37.4004;
+    static double dongAnLng = 126.9555;
+    static double appender = 0.002;
+
+    @Autowired
+    MasilService masilService;
+
+    @Autowired
+    MasilRepository masilRepository;
+
+    @Autowired
+    MasilPinRepository masilPinRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    User user;
+    String addressDepth1;
+    String addressDepth2;
+    String addressDepth3;
+    List<KakaoPoint> path;
+    String title;
+    int distance;
+    int totalTime;
+
+    @BeforeEach
+    void setUp() {
+        userRepository.save(createUser());
+        addressDepth1 = faker.address()
+            .state();
+        addressDepth2 = faker.address()
+            .city();
+        addressDepth3 = faker.address()
+            .streetName();
+        path = createPath(10000);
+        title = faker.lorem()
+            .maxLengthSentence(29);
+        distance = faker.number()
+            .numberBetween(1000, 1500);
+        totalTime = faker.number()
+            .numberBetween(10, 70);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {10, 0})
+    void 마실_생성을_성공한다(int expectedPinCount) {
+        // given
+        user = userRepository.save(createUser());
+        List<CreatePinRequest> pinRequests = createPinRequests(expectedPinCount, 10000);
+        CreateRequest request = new CreateRequest(addressDepth1, addressDepth2, addressDepth3,
+            "", path, title, null, distance, totalTime,
+            OffsetDateTime.now(), pinRequests, null, null);
+
+        // when
+        CreateResponse expected = masilService.create(user.getId(), request);
+
+        // then
+        Optional<Masil> optionalMasil = masilRepository.findById(expected.id());
+
+        assertThat(optionalMasil).isPresent();
+
+        Masil actual = optionalMasil.get();
```

Comment:
형님 테스트니까 이런 방식은 어떨까요?

```suggestion
        Masil actual = masilRepository
            .findById(expected.id())
            .orElseThrow();
```